### DISCUSSION TITLE
Open Banking Support

### DISCUSSION THREAD

**Discussion by sammcj** [ghostfolio/ghostfolio#2279] [Ideas]
It would be fantastic if this project could support the Open Banking standard.

This would allow it to integrate with Australian, New Zealand, UK (and other countries) financial institutions / banks via the standard Open Banking API to fetch up to date transactions, balances etc... securely.

- https://www.ausbanking.org.au/priorities/open-banking/
- https://www.openbanking.org.uk
- https://openbanking.ng

**Comment by dtslvr** [comment]
Hi @sammcj

Open Banking is also a hot topic in Switzerland (see https://blink.six-group.com). Do you have an idea how to try it out with simple proof of concept?

**Reply by sammcj** [reply]
Ah interesting, open + banking are not two words I think of when I think of banking in Switzerland 🤣.

I'm not sure how one goes about developing with it, but I did stumble across this the other day - https://github.com/LukePrior/Australian-Open-Banking-Data-Database

Also I know there are platforms/services such as https://www.basiq.io, https://frollo.com.au/open-banking/ etc... that offer open banking integration for developers (but they're paid / closed source).

**Comment by emmekappa** [comment]
It would be great an integration with GoCardless.

**Comment by john-frandsen** [comment]
The practical reality is that "Open Banking" splits into two quite different problems depending on region, and that shapes how hard a direct integration actually is.

**UK/EU (PSD2)** -- banks expose standardised APIs, but to call them *directly* you need an eIDAS QWAC + QSeal certificate (issued by a QTSP) and you have to register each set of bank APIs with the relevant National Competent Authority. The cert + onboarding overhead is brutal for a hobbyist or even a small company, which is essentially why aggregators exist at all -- they absorb the regulatory and cert burden and the per-bank quirks in exchange for one integration.

**Aggregators (the realistic path)** -- integrate once, get N banks. The main EU/UK ones worth knowing:
- **GoCardless / Nordigen** (same group now) -- a genuinely free Open Banking tier; you pay per requisition only if you want refresh beyond the ~90-day consent window. Best coverage and the cheapest entry point. This is what @emmekappa is asking about and it's a reasonable default to prototype against.
- **Tink** (Visa) -- polished, but skews enterprise-priced.
- **Yapily** -- API-only (no hosted consent UI), decent coverage.
- **Salt Edge / Enable Banking / FintechSystems** -- alternatives with different coverage/cost trade-offs; worth comparing bank lists country by country because coverage varies a lot.

For a **proof of concept** (answering @dtslvr's question above): the lowest-friction route is to grab a free Nordigen/GoCardless key, pick one bank from their *production* (not sandbox) list for wherever you test, and pull `/accounts/{id}/balances` + `/transactions`. Ghostfolio already has a generic import pipeline, so the smallest useful PoC is a small script that maps the aggregator's transactions response into Ghostfolio's existing manual-import shape (CSV/JSON) -- that proves the data mapping without touching the core, and gives you a feel for how messy real bank transaction data is before committing to a deeper integration.

**AU/NZ (CDR)** -- a separate standard (the ACCC Consumer Data Right) with its own participant-onboarding regime, closer to what @sammcj linked. Most EU aggregators don't cover CDR, which is why Basiq/Frollo come up in that market instead.

One thing worth deciding early: do you want **balances** (cash/account net-worth -- cheap, refresh-friendly, probably the highest-value for a portfolio tracker) or full **transactions** (heavier, tighter refresh limits, more bank-side throttling)? A balances-first MVP would cover the most common "I want my bank cash reflected in net worth" ask with the least complexity.
